src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/FutureFactory.cs

```diff
@@ -670,7 +671,7 @@ public Task<TResult> FromAsync(IAsyncResult asyncResult, Func<IAsyncResult, TRes
             if (Task.s_asyncDebuggingEnabled)
                 Task.AddToActiveTasks(t);
 
-            if (asyncResult.IsCompleted)
+            if (asyncResult.IsCompleted || OperatingSystem.IsBrowser())
```

Comment:
This also might not the solution, but annotating the API with  `[UnsupportedOSPlatform("browser")]`  would affect many related public APIs

Reply:
This is broken.  The change could result in the callback being run before the operation has actually completed.